### PR TITLE
Add cloud billing component

### PR DESCRIPTION
**Pull request by zugao**
## Summary

This PR adds cloud-billing to the component.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/class/defaults.yml [hunk 1]
```diff
@@ -612,3 +612,65 @@ parameters:
 
           # Generic config for objecstorage composite
           secretNamespace: ${crossplane:namespace}
+
+  billing:
+    namespace: appuio-cloud-reporting
+    clusterId: ${cluster:name}
+    salesOrderID: "" #Update once the fact is available in Leutenant
+    prometheus:
+      url: http://prometheus-operated.prometheus-system:9090/prometheus
```
Comment (Kidswiss) [inline component/class/defaults.yml:621]: I guess this will be changed to mimir later?

Comment (zugao) [reply component/class/defaults.yml:621]: I had it localhost, we can change later on when we deploy

File: component/class/defaults.yml [hunk 2]
```diff
@@ -612,3 +612,65 @@ parameters:
 
           # Generic config for objecstorage composite
           secretNamespace: ${crossplane:namespace}
+
+  billing:
+    namespace: appuio-cloud-reporting
```
[the remark this answers is not in the document]
Comment (zugao) [reply component/class/defaults.yml:617]: yeah makes sense

File: component/class/defaults.yml (hunk 1 above)
Comment (Kidswiss) [reply component/class/defaults.yml]: Good, just so it doesn't get missed :)

Also doesn't mimir require any sort of authentication?

**Review by Kidswiss — APPROVED** [review]

File: component/class/defaults.yml [hunk 3]
```diff
@@ -83,6 +88,55 @@ parameters:
...
+          cloudscale:
+            credentials:
+              stringData:
+                CLOUDSCALE_API_TOKEN: "?{vaultkv:${cluster:tenant}/${cluster:name}/billing-collector-cloudservices/cloudscale-token}"
+                KUBERNETES_SERVER_URL: "?{vaultkv:${cluster:tenant}/${cluster:name}/billing-collector-cloudservices/cluster-server}"
+                KUBERNETES_SERVER_TOKEN: "?{vaultkv:${cluster:tenant}/${cluster:name}/billing-collector-cloudservices/cluster-token}"
+                CONTROL_API_URL: "?{vaultkv:${cluster:tenant}/${cluster:name}/billing-collector-cloudservices/cloud-api-url}"
+                CONTROL_API_TOKEN: "?{vaultkv:${cluster:tenant}/${cluster:name}/billing-collector-cloudservices/cloud-api-token}"
+        # Unit of measure map "cloud service value": "odoo16 value"
+        uom: { }
```
Comment (Kidswiss) [inline component/class/defaults.yml:111]: Should we put the prod UOMs here? IMHO they are now defined and ready.

**Review by Kidswiss — CHANGES_REQUESTED** [review]

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**